src/Microsoft.IdentityModel.Logging/LogHelper.cs

```diff
@@ -384,7 +384,17 @@ private static object SanitizeSecurityArtifact(object arg)
 
             if (IdentityModelEventSource.LogCompleteSecurityArtifact && arg is ISafeLogSecurityArtifact)
                 return (arg as ISafeLogSecurityArtifact).UnsafeToString();
+            else if (arg is ISafeLogSecurityArtifact)
+            {
+                // We may later add a further flag which would log a best effort scrubbing of an artifact. E.g. JsonWebToken tries to remove the signature
+                // in the current implementation. Another flag may be added in the future to allow this middle path but for now, LogCompleteSecurityArtifact
+                // must be logged to emit any token part (other than specific claim values).
+                return string.Format(CultureInfo.InvariantCulture, IdentityModelEventSource.HiddenArtifactString, arg?.GetType().ToString() ?? "Null");
```

Comment:
No reason to check if arg is null as that is checked on line 382

Reply:
technically it's the toString return value not the arg. I agree this is unlikely to ever trigger though. This was an existing behavior, we can address outside this PR if important.

Reply:
What i am saying is 'arg' will never get here if null, every object has a type, ToString() is on every type.